Block/Catalog/Product/View.php

```diff
@@ -0,0 +1,249 @@
+ * Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
+ * WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF
+ * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+ * IN THE SOFTWARE.
+ *
+ * @author    Alma / Nabla SAS <contact@getalma.eu>
+ * @copyright 2018 Alma / Nabla SAS
+ * @license   https://opensource.org/licenses/MIT The MIT License
+ *
+ */
+
+namespace Alma\MonthlyPayments\Block\Catalog\Product;
+
+use Magento\Catalog\Block\Product\Context;
+use Alma\MonthlyPayments\Gateway\Config\Config;
+use Alma\MonthlyPayments\Helpers;
+use Magento\Framework\View\Element\Template;
+use Magento\Catalog\Model\Product;
+use Magento\Framework\Exception\LocalizedException;
+use Magento\Framework\Registry;
+
+class View extends Template
+{
+    /**
+     * @var Helpers\Eligibility
+     */
+    private $eligibilityHelper;
+
+    /**
+     * @var Config
+     */
+    private $config;
+
+    /**
+     * @var Registry
+     */
+    private $registry;
+
+    /**
+     * @var Product
+     */
+    private $product;
+
+    /**
+     * @var array
+     */
+    private $plans = array();
+
+    private const WIDGET_POSITION = 'payment/alma_monthly_payments/widget_position';
+
+    private const EXCLUDED_PRODUCT_TYPES = 'payment/alma_monthly_payments/excluded_product_types';
+
+    public $widgetContainer;
+
+    /**
+     * View constructor.
+     * @param Context $context
+     * @param Helpers\Eligibility $eligibilityHelper
+     * @param Registry $registry
+     * @param Config $config
+     * @param array $data
+     */
+    public function __construct(
+        Context $context,
+        Helpers\Eligibility $eligibilityHelper,
+        Registry $registry,
+        Config $config,
+        array $data = []
+    )
+    {
+        parent::__construct($context, $data);
+        $this->eligibilityHelper = $eligibilityHelper;
+        $this->config = $config;
+        $this->registry = $registry;
+    }
+
+    /**
+     * @return string
+     */
+    public function _toHtml()
+    {
+        return ($this->getNameInLayout() == $this->getConfig(SELF::WIDGET_POSITION)
+        && !$this->isExcluded() ? parent::_toHtml() : '');
+    }
+
+    public function isExcluded()
+    {
+        return in_array($this->getProductType(), $this->getExcludedProductType(), true);
+    }
+
+    /**
+     * Get config value.
+     *
+     * @param string $path
+     * @return string|null
+     */
+    public function getConfig($path)
+    {
+        return $this->_scopeConfig->getValue($path, \Magento\Store\Model\ScopeInterface::SCOPE_STORE);
+    }
+
+    /**
+     * @return false|string
+     */
+    public function getPlans()
```

Comment:
je ne suis pas fan des return type mixed ... soit l'array est empty, soit elle est peuplée (donc soit la chaîne est vide, soit elle ne l'est pas)

non ?

donc, idéalement `@rreturn string` tout cours :wink:

+ getPlans devrait return $this-plans (`array`) ... pour avoir une `json string`, on pourrait avoir une méthode `getJsonPlans`

Enfin, ce n'est pas le boulot du getter d'initialiser une propriété... je mettrai l'initialisation de `$this->plans` dans le constructeur ou un initialisateur dédié

Reply:
100% agree.
J'ai fixé dans ce sens (initPlans() et getJsonPlans() + j'ai aussi extrapolé ceci à getProduct en ajoutant un initProduct()